tools/flang1/flang1exe/semant.c

```diff
@@ -13286,7 +13286,7 @@ fixup_param_vars(SST *var, SST *init)
     sptr1 = get_param_alias_var(sptr, dtype);
   } else if (DTY(dtype) == TY_ARRAY) {
     ad = AD_DPTR(dtype);
-    if (AD_ASSUMSZ(ad) || AD_ADJARR(ad) || AD_DEFER(ad)) {
+    if (AD_ADJARR(ad) || AD_DEFER(ad)) {
```

Comment:
Flang would have given an error here for the following code previously, with this change the error does not appear.

```
program test
  integer, parameter :: asumed_sz(*) = 1
  print *, assumed_sz(1)
end program
```

Reply:
If the initializer in your example was changed to `(/1/)`, the shape should be implied to be `DIMENSION(1)`, but Flang currently would still complain incorrectly. gfortran can compile that case successfully.

Setting an assumed-shape array to a scalar integer has unclear semantics, and Flang should complain about this case.

It seems to me that the code should differentiate these cases and produce errors (or not) accordingly.